app/controllers/analyses/data_files_controller.rb

```diff
@@ -55,4 +55,8 @@ def create_params
   def decorate_data_file(data_file)
     AnalysisDataFileDecorator.decorate(data_file)
   end
+
+  def plant_trial
+    PlantTrial.visible(current_user.id).find(params[:plant_trial_id]) if params[:plant_trial_id].present?
```

Comment:
Maybe `find_by` and let `nil` pass to the Generator (which is able to deal with `nil`s anyway)?